gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckUnusedDependenciesTask.java

```diff
@@ -83,40 +83,38 @@ public final void checkUnusedDependencies() {
+        Set<ResolvedArtifact> possiblyUnusedArtifacts = declaredArtifacts.stream()
                 .filter(artifact ->
                         !necessaryArtifactsDeclaration.contains(BaselineExactDependencies.asString(artifact)))
                 .collect(Collectors.toSet());
         getLogger()
                 .debug(
                         "Possibly unused dependencies: {}",
-                        possiblyUnused.stream()
+                        possiblyUnusedArtifacts.stream()
                                 .map(BaselineExactDependencies::asString)
                                 .sorted()
                                 .collect(Collectors.toList()));
-        List<ResolvedArtifact> declaredButUnused = possiblyUnused.stream()
+        List<ResolvedArtifact> unusedArtifacts = possiblyUnusedArtifacts.stream()
                 .filter(artifact -> !shouldIgnore(artifact))
                 .sorted(Comparator.comparing(BaselineExactDependencies::asString))
                 .collect(Collectors.toList());
-        if (!declaredButUnused.isEmpty()) {
+        if (!unusedArtifacts.isEmpty()) {
             // TODO(dfox): don't print warnings for jars that define service loaded classes (e.g. meta-inf)
             StringBuilder builder = new StringBuilder();
             builder.append(String.format(
                     "Found %s dependencies unused during compilation, please delete them from '%s' or choose one of "
                             + "the suggested fixes:\n",
-                    declaredButUnused.size(), buildFile()));
-            for (ResolvedArtifact resolvedArtifact : declaredButUnused) {
+                    unusedArtifacts.size(), buildFile()));
+            for (ResolvedArtifact resolvedArtifact : unusedArtifacts) {
                 builder.append('\t')
                         .append(BaselineExactDependencies.asDependencyStringWithName(resolvedArtifact))
                         .append('\n');
 
-                // Suggest fixes by looking at all transitive classes, filtering the ones we have declarations on,
-                // and mapping the remaining ones back to the jars they came from.
-                ResolvedDependency dependency =
-                        BaselineExactDependencies.INDEXES.artifactsFromDependency(resolvedArtifact);
-                Set<ResolvedArtifact> didYouMean = dependency.getAllModuleArtifacts().stream()
-                        .filter(artifact ->
-                                BaselineExactDependencies.VALID_ARTIFACT_EXTENSIONS.contains(artifact.getExtension()))
+                // Suggest fixes by looking at all artifacts with the same module ID,
+                // filtering the ones we have declarations on, and mapping the remaining ones
+                // back to the jars they came from.
+                Set<ResolvedArtifact> didYouMean = BaselineExactDependencies.INDEXES
+                        .findArtifactsWithSameModuleId(resolvedArtifact)
```

Comment:
This doesn't seem equivalent to the previous implementation at all.

The intention of the existing code was to find implicit dependencies of the unused dependencies. I don't see how finding artifacts with the same module ID accomplishes that.